### DISCUSSION TITLE
Limit Task to One Instance per Worker in CeleryExecutor

### DISCUSSION THREAD

**Discussion by pedro-cf** [apache/airflow#42550] [Q&A] [answered]
I'm running Airflow with CeleryExecutor and I need to make sure that only a maximum of one instance of a specific task (taskA) runs on each worker at a time. Is there a simple way to achieve this without affecting the concurrency of other tasks?

I've explored options like max_active_tis_per_dag and task_concurrency, but it limits the task globally across all workers, which isn't what I want.

Any suggestions on how to achieve this?

**Comment by raphaelauv** [comment, marked as answer]
you can't with celery 

run the task in a separate execution platform ( like a kubernetes pod thanks to kubernetespodoperator )

**Reply by pedro-cf** [reply]
would be really cool if there was a task setting for `max_active_tis_per_worker`  :)

**Reply by raphaelauv** [reply]
the feature is not present in celery so it can't be done by airflow

**Reply by pedro-cf** [reply]
I have not tested this but what if I add the hostname to the task_id and set task_concurrency to 1? Would this effectivly work?
